src/drivers/ins/microstrain/modal_io_serial.cpp

```diff
@@ -0,0 +1,191 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "string.h"
+#include "modal_io_serial.hpp"
+
+ModalIoSerial::ModalIoSerial()
+{
+}
+
+ModalIoSerial::~ModalIoSerial()
+{
+	if (_uart_fd >= 0) {
+		uartClose();
+	}
+}
+
+int ModalIoSerial::uartOpen(const char *dev, speed_t speed)
+{
+	if (_uart_fd >= 0) {
+		PX4_ERR("Port in use: %s (%i)", dev, errno);
+		return -1;
+	}
+
+	/* Open UART */
+#ifdef __PX4_QURT
```

Comment:
Please don't include any platform defines (eg __PX4_QURT).

```suggestion
```

If you want/need a platform agnostic Serial wrapper (so it works on VOXL2/QURT) then you can use this Serial class. See drivers/gps or the voxl_esc for examples.

https://github.com/PX4/PX4-Autopilot/blob/main/platforms/common/include/px4_platform_common/Serial.hpp